modules/services/password-store-sync.nix

```diff
@@ -0,0 +1,78 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  meta.maintainers = with maintainers; [ pacien ];
+
+  options.services.password-store-sync = {
+    enable = mkEnableOption "Password store periodic sync";
+
+    frequency = mkOption {
+      type = types.str;
+      default = "*:0/5";
+      description = ''
+        How often to synchronise the password store git repository.
+        This value is passed to the systemd timer configuration as the onCalendar option.
+        See
+        <citerefentry>
+          <refentrytitle>systemd.time</refentrytitle>
+          <manvolnum>7</manvolnum>
+        </citerefentry>
+        for more information about the format.
+      '';
+    };
+  };
+
+  config = let
+    serviceCfg = config.services.password-store-sync;
+    programCfg = config.programs.password-store;
+
+    # from
+    # https://github.com/NixOS/nixpkgs/blob/release-19.03/nixos/modules/system/boot/systemd-lib.nix
+    # https://github.com/NixOS/nixpkgs/blob/release-19.03/nixos/modules/system/boot/systemd.nix
+    makeJobScript = let
+      shellEscape = s: (lib.replaceChars [ "\\" ] [ "\\\\" ] s);
+      mkScriptName = s: "unit-script-" + (lib.replaceChars [ "\\" "@" ] [ "-" "_" ] (shellEscape s) );
+      mkScriptContent = script: ''
+        #! ${pkgs.runtimeShell} -e
+        ${script}
+      '';
+    in name: text: toString (pkgs.writeTextFile {
```

Comment:
I think you can you use [writeShellScript](https://github.com/nixos/nixpkgs/blob/3f880f187d0bdf1f1c98bb3cfee3b1b092508c45/pkgs/build-support/trivial-builders.nix#L145-L168):
```nix
makeJobScript = name: text:
  let shellEscape = s: (lib.replaceChars [ "\\" ] [ "\\\\" ] s);
      mkScriptName = s: "unit-script-" + (lib.replaceChars [ "\\" "@" ] [ "-" "_" ] (shellEscape s) );
  in writeShellScript (mkScriptName name) text;
```

Reply:
`writeShellScript` doesn't seem to be available on 19.03 so I used `writeScript` instead to keep backward compatibility.